modules/javafx.graphics/src/main/java/com/sun/prism/Image.java

```diff
@@ -650,9 +650,9 @@ public Image promoteByteRgbToByteBgra() {
                                                   (IntBuffer) pixelBuffer);
                     break;
             }
-        }
-        if (pixelaccessor != null && pixelScale != 1.0f) {
-            pixelaccessor = new ScaledAccessor<>(pixelaccessor, pixelScale);
+            if (pixelScale != 1.0f) {
+                pixelaccessor = new ScaledAccessor<>(pixelaccessor, pixelScale);
```

Comment:
is that really correct? I think you should not overwrite save the scaled pixelAccessor in the instance variable - if I read that correct with this change the following happens: if I call getPixelAccessor() 3 times I get a ScaledAccessor wrapping a ScaleAccessor wrapping a ScaledAccessor, wrapping one of ByteAccess, ByteRgbAccess, IntAccess

Reply:
No, that’s what used to happen before this fix.

The fix moves the wrapping of the pixelAccessor with a ScaledPixelAxcessor inside the null check so it’s now only done once.

Reply:
ah I missed that the lines moved up inside the first if-check